cpg-core/src/main/java/de/fraunhofer/aisec/cpg/graph/TypeManager.java

```diff
@@ -546,23 +547,35 @@ private Set<Ancestor> getAncestors(RecordDeclaration recordDeclaration, int dept
     return ancestors;
   }
 
-  public boolean isSupertypeOf(Type superType, Type subType, ScopeProvider provider) {
+  public boolean isSupertypeOf(Type superType, Type subType, MetadataProvider provider) {
```

Comment:
Is any MetadataProvider allowed and expected here. I not I would either restrict the type or if there is no specific type then name it to something descriptive like `scopeProvider`

Reply:
Basically, we need a metadata provider that is ideally a language provider and a scope provider. Unfortunately we cannot scope it like that in Java (I think it may be possible in Kotlin), so I chose the basic metadata provider class.